checkers/rules/rules.go

```diff
@@ -631,3 +631,48 @@ func exposedSyncMutex(m dsl.Matcher) {
 		Where(m["x"].Text.Matches(`^\p{Lu}`)).
 		Report("don't embed *sync.RWMutex")
 }
+
+//doc:summary Detects suspicious duplicated sub-expressions
+//doc:tags    diagnostic
+//doc:before  xs[i] < xs[i]
+//doc:after   xs[i] < xs[j]
+func dupSubExpr(m dsl.Matcher) {
+	m.Match(`$x || $x`).
+		Report(`suspicious identical left and right expression near ||`)
+	m.Match(`$x && $x`).
+		Report(`suspicious identical left and right expression near &&`)
+	m.Match(`$x | $x`).
+		Report(`suspicious identical left and right expression near |`)
+	m.Match(`$x & $x`).
+		Report(`suspicious identical left and right expression near &`)
+	m.Match(`$x ^ $x`).
+		Report(`suspicious identical left and right expression near ^`)
+	m.Match(`$x < $x`).
+		Report(`suspicious identical left and right expression near <`)
+	m.Match(`$x > $x`).
+		Report(`suspicious identical left and right expression near >`)
+	m.Match(`$x &^ $x`).
+		Report(`suspicious identical left and right expression near &^`)
+	m.Match(`$x % $x`).
+		Report(`suspicious identical left and right expression near %`)
+
+	// TODO
```

Comment:
What exactly this TODO means? :)

Reply:
AFAIR there were failures in expressions below...